jsx/Form.js

```diff
@@ -178,16 +178,12 @@ class SearchableDropdown extends React.Component {
       let options = this.props.options;
       if (Object.values(options).indexOf(value) === -1) {
         // empty string out both the hidden value as well as the input text
-        document.querySelector(`input[name="${this.props.name + '_input'}"]`).value = '';
+        this.props.onUserInput(this.props.name + 'Temp', '');
```

Comment:
> I don't understand what this "Temp" thing is all about in this PR.
- dave

Reply:
there needs to be an intermediary "temporary" value for the user's input to be stored in as they're typing. couldn't think of a better naming convention. open to any suggestions

Reply:
I still don't really understand.. why can't it just use state?

Reply:
i think it's simpler to keep it stateless and to just rely on the user component's state, since this tends to be the norm currently for our form components.
this way also fixes another issue we were having on CCNA: in media, after a successful upload `this.state.formData` should clear out, but because the text of the input was not kept in `formData`, the text inside the Site input would remain.

Reply:
I agree that Form.js components should generally remain stateless, but maybe this should be an exception? I'm not really sure that the user component's state should be holding on to this temp value. Is the implication that the temp value will be held in the parent components `formData` (or equivalent) object? It also seems like this solution necessitates the parent component to pass both a value and temp value down as a prop. If the parent component doesn't need this information, I would opt for it to be held in state in this component.

Reply:
> I agree that Form.js components should generally remain stateless, but maybe this should be an exception? I'm not really sure that the user component's state should be holding on to this temp value. Is the implication that the temp value will be held in the parent components `formData` (or equivalent) object? It also seems like this solution necessitates the parent component to pass both a value and temp value down as a prop. If the parent component doesn't need this information, I would opt for it to be held in state in this component.

you have a good point. the only benefit of keeping it in the parent component's state would be, for example, in media's upload form, upon successful upload formData gets cleared out, but the text inside the input would remain. i'll tinker around with keeping it in `SearchableDropdown`'s state



> @davidblader @driusan @HenriRabalais Why not just set a default value for the this.props.name? You could set the default value to be 'Temp' and not need to add the + 'Temp' which looks ugly.

can you elaborate? `this.props.name` is meant to be the key in the formData object that gets passed to the server. i'm not sure we want to mess around with that.

Reply:
I added the "Needs Work" tag until this is sorted out.. (it's not the most appropriate tag, but I couldn't think of a better one..)

Reply:
@davidblader Oh I see now. I guess another prop could be made to replace the string and just set the defaultProps for the new prop to Temp. I'm not sure now.